src/DotNetEnv/Parser.cs

```diff
@@ -10,22 +14,16 @@ private static bool IsComment(string line)
         private static string RemoveInlineComment(string line)
         {
             int pos = line.IndexOf('#');
-            if (pos == -1)
-                return line;
-
-            return line.Substring(0, pos);
+            return pos >= 0 ? line.Substring(0, pos) : line;
         }
 
         private static string RemoveExportKeyword(string line)
         {
-            line = line.TrimStart(' ');
-            if (!line.StartsWith("export "))
-                return line;
-
-            return line.Substring(7);
+            Match match = ExportRegex.Match(line);
+            return match.Success ? line.Substring(match.Length) : line;
```

Comment:
The old implementation of this method *always* removed leading whitespace on keys.  This meant that when the `trimWhitespace` param was `false` and should retain all whitespace, the key would still have no leading spaces (verified in `LoadArgsTest` second assert).